pkg/kwok/controllers/node_controller.go

```diff
@@ -483,6 +488,18 @@ func (c *NodeController) computePatchNode(node *corev1.Node, tpl string) ([]byte
 	})
 }
 
+// putNodeInfo puts node info (HostIPs)
+func (c *NodeController) putNodeInfo(node *corev1.Node) {
```

Comment:
https://github.com/kubernetes/kubernetes/blob/1d02d014e8c1f0de84b0b58b2165548182815320/pkg/util/node/node.go#L67-L104

We should refer to the logic of how Kubelet handles addresses.

Reply:
Thank you. I added the re-sort and also refer to v4 before v6.

Reply:
Got timely test failure. Do you think is it related to re-sort logic when putting the nodeInfo? Should I optimize that? I think the kubelet logic needs only first ipv4 and first ipv6 (if exist) sorting from internal.

Reply:
According to `'.status.addresses.[] | select( .type == "InternalIP" ) | .address'`, 
should we optimize the logic to get only first ipv4/ipv6 which is InternalIP? Or should we keep all re-sorting logic?

Reply:
Both should be feasible now, and my personal preference is to keep the sorting and the two IPs.